lib/s3_helper.rb

```diff
@@ -40,6 +40,24 @@ def initialize(s3_bucket_name, tombstone_prefix = '', options = {})
       end
   end
 
+  def self.build_from_config(use_db_s3_config: false, for_backup: false, s3_client: nil)
+    bucket, options =
+      if use_db_s3_config
+        [
+          for_backup ? SiteSetting.s3_backup_bucket.downcase : SiteSetting.s3_upload_bucket.downcase,
+          S3Helper.s3_options(SiteSetting)
+        ]
+      else
+        [
+          for_backup ? GlobalSetting.s3_backup_bucket.downcase : GlobalSetting.s3_bucket.downcase,
+          S3Helper.s3_options(GlobalSetting)
+        ]
+      end
+    options[:client] = s3_client if s3_client.present?
+
+    S3Helper.new(bucket, '', options)
+  end
```

Comment:
```suggestion
  def self.build_from_config(use_db_s3_config: false, for_backup: false, s3_client: nil)
    klass = use_db_s3_config ? SiteSetting : GlobalSetting
    
    bucket, options = [
      for_backup ? klass.s3_backup_bucket.downcase : klass.s3_upload_bucket.downcase,
      S3Helper.s3_options(klass)
    ]
    
    options[:client] = s3_client if s3_client.present?

    S3Helper.new(bucket, '', options)
  end
```

Reply:
Unfortunately this does not _quite_ work because the S3 bucket property is slightly different between global + site:

![image](https://user-images.githubusercontent.com/920448/140009103-d040cba5-c1cf-40dd-a550-bd6bb1c2d8b5.png)

Reply:
Will try to clean it up a bit though.